### ISSUE TITLE
Video Uploads are extremely slow

### ISSUE THREAD

**Issue by hashweather** [fastify/busboy#104]
### Prerequisites

- [X] I have written a descriptive issue title
- [X] I have searched existing issues to ensure the issue has not already been raised

### Issue

I am not sure what may be the issue but when uploading short length videos of less than 2MB the video upload is blazing fast.

However, the moment I tried to upload a 20MB video, the time it takes before busboy starts to process the video is so so long.

Here is my code snippet:
```

busboy.on('file', async function (fieldname, file, filename, encoding, mimetype) {

             console.log("Video Reached Busboy"); //This happened very fast

            if (!unProcessedFilePaths.includes(filename)) {

                let fileType = await FileService.getFileType(filename);
                let transferredBytes = 0;

                console.log(`On Data about to be called`); 

                file.on('data', async (data) => {
                    console.log(`On Data is been called`);   //This is called only after about 2mins for a 20MB video. Weird

``` 

Please advise.

Thank you.

**Comment by kibertoad** [comment]
as a quick thought, you may want to experiment with highWaterMark size

**Comment by hashweather** [comment]
@kibertoad Thanks for that insight. However, I'm not sure what the optimal sizes maybe.

Do you mind recommending sizes that will make my videos upload faster?

Thank you

**Comment by kibertoad** [comment]
Never had to do this kind of tweaking myself. Try increasing them significantly from defaults and measure whether it makes any perf difference for your use-cases.

**Comment by Uzlopak** [comment]
limits.fileSize is limitting the allowed uploadable filesize. That wont help your case. 

You have to take care of the highWateMarks of the streams. i think the default value is 16 kb. So imagine chunking 20 mb into 16 kb blocks, results in a big overhead.

Maybe increase it to 512 kb. Would be then 40 chunks for 20 mb.

**Comment by hashweather** [comment]
Thank you so much for the detailed breakdown. 

However for the 6e+8 I think it's just 600MB because I actually crossed checked with Google early on before setting it unless maybe I'm wrong. 

Here is what the busboy documentation says about the file size:

`fileSize - integer - For multipart forms, the max file size (in bytes) (Default: Infinity).`

But here is the result I got from Google
<img width="669" alt="Screenshot 2022-12-06 at 7 31 01 PM" src="https://user-images.githubusercontent.com/48493864/205993074-163f02c5-c110-4484-88ec-955a5e535e15.png">


Or am I wrong?

Thank you once again
